subsys/net/ip/icmpv4.c

```diff
@@ -26,11 +26,93 @@
 
 static sys_slist_t handlers;
 
+struct net_icmp_hdr *net_icmpv4_set_hdr(struct net_pkt *pkt,
+					struct net_icmp_hdr *hdr)
+{
+	struct net_buf *frag = pkt->frags;
+	struct net_icmp_hdr *icmp_hdr;
+	u16_t pos;
+
+	icmp_hdr = net_pkt_icmp_data(pkt);
+	if (net_check_icmp_header(pkt, icmp_hdr)) {
```

Comment:
May go even more detailed and "fully correct", like net_header_fits_frag(), but that gets long and IMHO even above is good improvement.